application/core/Catalog_controller.php

```diff
@@ -24,33 +24,77 @@ function _render($file)
 		$this->load->view($file, $this->data);
 	}
 
-	function _format_pagination($first_page, $page_count, $call_function = 'get_results')
+	function _format_pagination($current_page, $page_count, $call_function = 'get_results')
 	{
-		$pages = $this->_build_pagination_array($first_page, $page_count);
-
-		$html = '<a class="page-number first" data-page_number="1" href="#" data-call_function="' . $call_function . '">&laquo;</a>';
-
-		foreach ($pages as $key => $page)
+        $page_count = intval($page_count); // It's a float, but we want ints. (Causes problems when generating ranges.)
```

Comment:
I wasn't sure what the PHP way of doing this is, sorry.

`_build_pagination_array` normally returns an array of integers, but if `$current_page` is near the end, then I think the `min(($first_page + $num_links), max($page_count, 1))` ends up being a float instead of an int, and so the returned range is a range of floats rather than a range of ints.

I didn't notice it until I tried to do a `===` on it to compare `$page` and `$page_count` :slightly_smiling_face:

Reply:
I'm no PHP expert either, we'll just have to live with this weirdness.